--- setup.py-orig	2013-12-10 19:13:27.000000000 -0500
+++ setup.py	2013-12-10 22:36:22.000000000 -0500
@@ -258,17 +258,19 @@
     # force x64 because Snow Leopard's native Python is 64-bit
     # scons arch=x64 library=static
 
-    include_dirs += [
-        "/opt/local/include", # MacPorts$ sudo port install boost
-        "/usr/local/include", # HomeBrew$ brew install boost
-    ]
-
-    library_dirs += [
-        "/opt/local/lib", # MacPorts$ sudo port install boost
-        "/usr/local/lib", # HomeBrew$ brew install boost
-    ]
-
-    libraries += ["boost_python-mt"]
+    #include_dirs += [
+    #    "/opt/local/include", # MacPorts$ sudo port install boost
+    #    "/usr/local/include", # HomeBrew$ brew install boost
+    #]
+
+    #library_dirs += [
+    #    "/opt/local/lib", # MacPorts$ sudo port install boost
+    #    "/usr/local/lib", # HomeBrew$ brew install boost
+    #]
+
+    libraries += ["boost_python-mt-d"]
+    include_dirs += [ os.path.join(BOOST_HOME, 'include'), ]
+    library_dirs += [ os.path.join(BOOST_HOME, 'lib'), ]
 
     is_64bit = math.trunc(math.ceil(math.log(sys.maxint, 2)) + 1) == 64 # contribute by viy
 
@@ -280,6 +282,10 @@
         os.environ['ARCHFLAGS'] = '-arch i386'
         macros += [("V8_TARGET_ARCH_IA32", None)]
 
+    common_args = ["-stdlib=libstdc++"]
+    extra_compile_args += ["-Wdeprecated-writable-strings", "-Wno-deprecated-declarations"] + common_args
+    extra_link_args += common_args
+
 else:
     print "ERROR: unsupported OS (%s) and platform (%s)" % (os.name, sys.platform)
 
@@ -347,7 +353,7 @@
         print "INFO: we will try to use the system 'svn' command to checkout/update V8 code"
 
     if update_code:
-        args = ["svn", "up", V8_HOME]
+        args = ["svn", "update", V8_HOME]
     else:
         os.makedirs(V8_HOME)
 
@@ -457,7 +463,7 @@
 
         options = ' '.join(["%s=%s" % (k, v) for k, v in options.items()])
 
-        cmdline = "%s %s %s.%s" % (MAKE, options, arch, mode)
+        cmdline = "%s -j8 %s %s.%s" % (MAKE, options, arch, mode)
 
         exec_cmd(cmdline, "build v8 from SVN")
 
