--- v8-readonly/src/parser.h	2012-10-02 23:27:44.000000000 +0200
+++ v8/src/parser.h	2012-10-02 23:27:32.000000000 +0200
@@ -453,7 +453,7 @@
   // Code::Flags uses only the low 17 bits of num-parameters to
   // construct a hashable id, so if more than 2^17 are allowed, this
   // should be checked.
-  static const int kMaxNumFunctionParameters = 32766;
+  static const int kMaxNumFunctionParameters = 65534; //32766
   static const int kMaxNumFunctionLocals = 65535;
 
   enum Mode {
