--- v8-readonly/src/log.h	2012-10-02 23:27:44.000000000 +0200
+++ v8/src/log.h	2012-10-02 23:24:17.000000000 +0200
@@ -80,6 +80,7 @@
 #undef LOG
 #define LOG(isolate, Call)                          \
   do {                                              \
+    isolate->InitializeLoggingAndCounters();        \
     v8::internal::Logger* logger =                  \
         (isolate)->logger();                        \
     if (logger->is_logging())                       \
