--- v8-readonly/src/log.h	2011-11-15 16:47:15.000000000 +0100
+++ v8/src/log.h	2011-11-15 16:44:30.000000000 +0100
@@ -81,6 +81,7 @@
 #undef LOG
 #define LOG(isolate, Call)                          \
   do {                                              \
+    isolate->InitializeLoggingAndCounters();        \
     v8::internal::Logger* logger =                  \
         (isolate)->logger();                        \
     if (logger->is_logging())                       \
